config/mbed/CMakeLists.txt

```diff
@@ -0,0 +1,279 @@
+chip_gn_arg_bool  ("is_debug"                             CONFIG_DEBUG)
+chip_gn_arg_bool  ("chip_build_tests"                     CONFIG_CHIP_BUILD_TESTS)
+chip_gn_arg_bool  ("chip_build_libshell"                  CONFIG_CHIP_LIB_SHELL)
+chip_gn_arg_bool  ("chip_with_platform_mbedtls"           CONFIG_CHIP_WITH_EXTERNAL_MBEDTLS)
+chip_gn_arg_bool  ("chip_bypass_rendezvous"               CONFIG_CHIP_BYPASS_RENDEZVOUS)
+
+if (BOARD STREQUAL "native_posix")
+    chip_gn_arg_string("target_cpu" "x86")
+elseif (BOARD STREQUAL "native_posix_64")
+    chip_gn_arg_string("target_cpu" "x64")
+endif()
+
+file(GENERATE OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/args.gn CONTENT ${CHIP_GN_ARGS})
+
+# ==============================================================================
+# Define 'chip-gn' target that builds CHIP library(ies) with GN build system
+# ==============================================================================
+ExternalProject_Add(
+    chip-gn
+    PREFIX                  ${CMAKE_CURRENT_BINARY_DIR}
+    SOURCE_DIR              ${CHIP_ROOT}
+    BINARY_DIR              ${CMAKE_CURRENT_BINARY_DIR}
+    CONFIGURE_COMMAND       gn --root=${CHIP_ROOT}/config/mbed/chip-gn gen --check --fail-on-unused-args ${CMAKE_CURRENT_BINARY_DIR}
+    BUILD_COMMAND           ninja
+    INSTALL_COMMAND         ""
+    BUILD_BYPRODUCTS        ${CHIP_LIBRARIES}
+    BUILD_ALWAYS            TRUE
+    USES_TERMINAL_CONFIGURE TRUE
+    USES_TERMINAL_BUILD     TRUE
+)
+
+# ==============================================================================
+# Define 'chip' target that exposes CHIP headers & libraries to the application
+# ==============================================================================
+mbed_interface_library_named(chip)
+target_compile_definitions(chip INTERFACE CHIP_HAVE_CONFIG_H)
+target_include_directories(chip INTERFACE
+    ${CHIP_ROOT}/src
+    ${CHIP_ROOT}/src/app/server
+    ${CHIP_ROOT}/src/app/util
+    ${CHIP_ROOT}/src/include
+    ${CHIP_ROOT}/src/lib
+    ${CHIP_ROOT}/src/lib/core
+    ${CHIP_ROOT}/third_party/nlassert/repo/include
+    ${CMAKE_CURRENT_BINARY_DIR}/gen/include
+    ${CMAKE_CURRENT_BINARY_DIR}/gen/third_party/connectedhomeip/src/app/include
+)
+target_link_directories(chip INTERFACE ${CMAKE_CURRENT_BINARY_DIR}/lib)
+target_link_libraries(chip INTERFACE -Wl,--start-group ${CHIP_LIBRARIES} -Wl,--end-group)
+add_dependencies(chip chip-gn)
+
+# ==============================================================================
+# Define mbed target configuration according to CHIP component usage
+# ==============================================================================
+# CHIP includes path
+list(APPEND CHIP_INCLUDES)
+
+# CHIP defines
+list(APPEND CHIP_DEFINES)
+
+
+list(APPEND CHIP_INCLUDES 
+    ${CHIP_ROOT}/config/mbed/mbedtls
+    ${CHIP_ROOT}/src/platform/mbed/posix/include
+    ${CHIP_ROOT}/src/platform/mbed/net/include
+)
+
+list(APPEND CHIP_DEFINES 
+    __LINUX_ERRNO_EXTENSIONS__=1
```

Comment:
What does this do?

Is errno task local in Mbed ?

Reply:
We need that define to access all the error codes required by our BSD socket implementation.

Reply:
Is errno task local in Mbed ?

Reply:
I'm not sure to understand what you mean by _task_. The `errno` variable (not really a variable by default with newlib) should be global to the application and there is mechanism in place in newlib and Mbed to handle concurrent access.